specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,260 @@
+        verticalinset="true"
+    >
+      This overlaps the anchor and extends above or below it depending on available space.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region renders above or below the anchor based on available space and is sized to match the available.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchor="anchor"
+        verticalPositioningMode="dynamic"
+        verticalscalingenabled="true"
+    >
+       This region renders above or below the anchor based on available space and sizes itself to match.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders below the anchor until that space is less than 100px.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchor="anchor"
+        verticalpositioningmode="dynamic"
+        verticaldefaultposition="below"
+        verticalthreshold="100"
+    >
+      This shows renders below the anchor as long as there is at least 100px available there.
+    </anchored-region>
+    ...stuff...
+</div>
+```
+
+The dimensions of the anchored region will match the dimensions of the content unless scaling is enabled on a particular axis (verticalscalingenabled & horizontalscalingenabled) in which case it will fill all available space between the anchor and viewport.
+
+The component allows users to set a "Positioning Mode" on each axis which defines how the component will behave:
+- 'uncontrolled':  The anchored region will appear as it normally would in document flow.
+- 'locktodefault': The anchored region will always be placed in the specified default position regardless of available space.  For example a menu that always opens upwards.
+- 'dynamic': The anchored region is placed relative to the anchor element based on how much space is available between it and the viewport.  For example a menu that opens opens upwards when it is near the bottom of the page, and downwards when near the top.
+- 'onetime': As the 'dynamic' option, but position is not changed after initial placement.
+
+### API
+
+*Component name:*
+- `fast-anchored-region`
+
+*Attributes:*
+- anchor - The html id of the HTMLElement used as the anchor around which the positioning region is placed.  This must be set for the component's positioning logic to be active.
+
+- viewport - Used to identify the HTMLElement used as the viewport the component uses to determine available layout space around the anchor element. Possible values are the id of the desired viewport element, '#document' or '#parent'.  The default value is '#parent'.
+
+- horizontalPositioningMode - Can be 'uncontrolled', 'locktodefault', 'dynamic' or 'onetime'.  Default is 'uncontrolled'.
+- horizontalDefaultPosition - Can be 'start', 'end', 'left', 'right' or 'unset'.  Default is 'unset'
```

Comment:
I would also characterize this as "Next steps" because I believe default browser behavior can be reflected here, same as the verticalDefaultPosition where it is assumed to be a dropdown menu, aligned to the left of the anchor.

Reply:
I'd argue this is core functionality that should be part of the first iteration.